src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -2910,4 +2910,59 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
             let Foo { r#field: _ } = f;
         }
     """)
+
+    @MockRustcVersion("1.40.0-nightly")
+    fun `test unstable feature E0658 1`() = checkErrors("""
+        mod a {
+            #[unstable(feature = "aaa")]
+            pub struct S {
+                #[unstable(feature = "bbb", reason = "foo")] pub field: i32
+            }
+        }
+        
+        use a::<error descr="`aaa` is unstable [E0658]">S</error>;
+        
+        #[unstable(feature = "ddd")]
+        impl <error descr="`aaa` is unstable [E0658]">S</error> {
+            #[unstable(feature = "ccc", reason = "bar \
+                baz")]
+            fn foo(self) -> Self {}
+        }
+        
+        fn main() {
+            let x = <error descr="`aaa` is unstable [E0658]">S</error> { field: 0 };
+            x.<error descr="`bbb` is unstable: foo [E0658]">field</error>;
+            x.<error descr="`ccc` is unstable: bar baz [E0658]">foo()</error>;
```

Comment:
You still highlight too wide range because it contains all method params
Should be 
```
x.<error descr="`ccc` is unstable: bar baz [E0658]">foo</error>();
```

`org.rust.lang.core.psi.ext.RsReferenceElement#getReferenceNameElement` may be helpful here but looks like we don't want to highlight `Self` paths so probably, you need to check element type of `getReferenceNameElement` is identifier

Reply:
Fixed.